### PR TITLE
Pickle da.argwhere and da.count_nonzero

### PR DESCRIPTION
**Pull request by crusaderky**
Fix serialization of da.argwhere and da.count_nonzero:

```
_pickle.PicklingError: Can't pickle <ufunc '_isnonzero_vec (vectorized)'>: attribute lookup _isnonzero_vec (vectorized) on __main__ failed
```

### CODE REVIEW

File: dask/array/routines.py [hunk 1]
```diff
@@ -2049,31 +2049,22 @@ def choose(a, choices):
     return elemwise(variadic_choose, a, *choices)
 
 
-def _isnonzero_vec(v):
-    return bool(np.count_nonzero(v))
+_isnonzero_vec = np.vectorize(lambda v: bool(np.count_nonzero(v)), otypes=[bool])
```
Comment (phofl) [inline dask/array/routines.py:2052]: Why the rewrite here?

Comment (crusaderky) [reply dask/array/routines.py:2052]: 1. This looked slightly more compact and readable IMHO
2. Best practice of not changing the type of a variable after initial assignment

Ultimately it is cosmetic.

Comment (crusaderky) [reply dask/array/routines.py]: reverted

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx

**Pull request merged**